src/Extensions/Diagnostics.HealthChecks/HealthChecksBuilderExtensions.cs

```diff
@@ -40,9 +42,73 @@ public static class HealthChecksBuilderExtensions
 			Func<HttpResponseMessage, HealthCheckResult, HealthCheckResult>? additionalCheck = null,
 			params KeyValuePair<string, object>[] reportData)
 		{
+			scheme = scheme == null
+					? Uri.UriSchemeHttp
+					: Uri.CheckSchemeName(scheme)
+						? scheme
+						: throw new ArgumentException("Invalid uri scheme", nameof(scheme));
+
+			if(!Uri.TryCreate(relativePath, UriKind.Relative, out Uri? result) || result.IsAbsoluteUri)
+			{
+				throw new ArgumentException("relativePath is not valid or can't be an Absolute uri", nameof(relativePath));
+			}
+
+			Func<UriBuilder, HttpRequestMessage> httpRequest = uriBuilder =>
+			{
+				uriBuilder.Path = relativePath;
+				uriBuilder.Scheme = scheme;
+				HttpRequestMessage requestMessage = new (method ?? HttpMethod.Get, uriBuilder.Uri);
+
+				if (headers != null)
+				{
+					foreach (KeyValuePair<string, IEnumerable<string>> pair in headers)
+					{
+						if(!requestMessage.Headers.TryAddWithoutValidation(pair.Key, pair.Value))
+						{
+							string errorMessage = string.Format(
+								CultureInfo.InvariantCulture,
+								"Cannot add request header with name '{0}' value '{1}' for health check '{2}'.",
+								pair.Key,
+								pair.Value,
+								name);
+
+							throw new ArgumentException(errorMessage, nameof(headers));
+						}
+					}
+				}
+
+				return requestMessage;
+			};
+
+			return builder.AddHttpEndpointCheck(name, endpointName, httpRequest, expectedStatus, additionalCheck, reportData);
+		}
+
+		/// <summary>
+		/// Add http endpoint health check
+		/// </summary>
+		/// <param name="builder">health checks builder</param>
+		/// <param name="name">name of the health check</param>
+		/// <param name="endpointName">name of the endpoint to check</param>
+		/// <param name="httpRequestMessageBuilder">action that will return the http request message</param>
+		/// <param name="expectedStatus">response status code that considered healthy, default to 200(OK)</param>
+		/// <param name="additionalCheck">action that would be called after getting response, function should return new result object that would be reported</param>
+		/// <param name="reportData">additional properties that will be attached to health check result, for example escalation info</param>
+		public static IHealthChecksBuilder AddHttpEndpointCheck(
+			this IHealthChecksBuilder builder,
+			string name,
+			string endpointName,
+			Func<UriBuilder, HttpRequestMessage> httpRequestMessageBuilder,
+			HttpStatusCode? expectedStatus = null,
+			Func<HttpResponseMessage, HealthCheckResult, HealthCheckResult>? additionalCheck = null,
```

Comment:
Is it worth it making this async friendly?

Reply:
For `httpRequestMessageBuilder` I hope it's not needed since it's mostly for configuring additional properties of the request. And the fact that we are invoking it in this method would force us to make it async also, so no.

But it's a really good idea for `additionalCheck` since you might need to get content there and all of the methods for it are async. It's a bit out of the scope of this PR, but definitely something we need to do. Thank you.

@vh-vahan  I'll live it up to you to include the change in this PR or I'll create an issue and we can handle it separately.

Reply:
Thanks. Good point about breaking change. I'll send a note about it.